### PR TITLE
Document volume expansion and volume claim immutability

### PR DESCRIPTION
**Pull request by sebgl**
This commit modifies the existing documentation in order to account
for the volume expansion feature introduced in https://github.com/elastic/cloud-on-k8s/pull/3752.

With those changes, it is now possible to resize storage requests if the storage size
allows it.
Every other change is forbidden.

The doc also mentions how Pods may need to be restarted if the volume driver
does not support online expansion.

I added lower-level titles to better structure the volume claim templates page.

Fixes https://github.com/elastic/cloud-on-k8s/issues/325.

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 1]
```diff
@@ -30,7 +33,17 @@ spec:
...
 
-IMPORTANT: Depending on the Kubernetes configuration and the underlying file system, some persistent volumes <<{p}-orchestration-limitations,cannot be resized after they are created>>. When you define volume claims, consider future storage requirements and make sure you have enough space to support the expected growth.
+[float]
+== Updating the volume claim settings
+
+If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in the volumeClaimTemplates. ECK will update the existing PersistentVolumeClaims accordingly, and recreate the StatefulSet automatically. If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem will be resized online, and does not require restarting the Elasticsearch process nor re-creating the Pods. If the volume driver does not support `ExpandInUsePersistentVolumes`, it is necessary to manually delete the Pods after the resize, to be recreated automatically with the expanded filesystem.
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:39]:
```suggestion
If the storage class allows link:https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/[volume expansion], you can increase the storage requests size in the volumeClaimTemplates. ECK will update the existing PersistentVolumeClaims accordingly, and recreate the StatefulSet automatically. If the volume driver supports `ExpandInUsePersistentVolumes`, the filesystem is resized online, without the need of restarting the Elasticsearch process, or re-creating the Pods. If the volume driver does not support `ExpandInUsePersistentVolumes`, Pods must be manually deleted after the resize, to be recreated automatically with the expanded filesystem.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc [hunk 2]
```diff
@@ -30,7 +33,17 @@ spec:
...
+
+Any other changes in the volumeClaimTemplates, such as changing the storage class or decreasing the volume size, is forbidden. In order to achieve those, you can create a new nodeSet with different settings, and remove the existing nodeSet. In practice, that's equivalent to renaming the existing nodeSet while modifying its claim settings in a single update. ECK ensures data stored on Pods of the deleted nodeSet is migrated to other nodes before removing Pods of the deleted nodeSet.
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/volume-claim-templates.asciidoc:41]:
```suggestion
Any other changes are forbidden in the volumeClaimTemplates, such as changing the storage class or decreasing the volume size. To make these changes, you can create a new nodeSet with different settings, and remove the existing nodeSet. In practice, that's equivalent to renaming the existing nodeSet while modifying its claim settings in a single update. Before removing Pods of the deleted nodeSet, ECK makes sure that data is migrated to other nodes.
```

**Review by alaudazzi — APPROVED** [review]
Minor editing suggestions, otherwise LGTM.

**Review by barkbay — APPROVED** [review]
lgtm

**Pull request merged**